src/psyclone/psyir/backend/fortran.py

```diff
@@ -876,16 +889,17 @@ def gen_decls(self, symbol_table, is_module_scope=False):
         SymbolTable.
 
         :param symbol_table: the SymbolTable instance.
-        :type symbol: :py:class:`psyclone.psyir.symbols.SymbolTable`
+        :type symbol_table: :py:class:`psyclone.psyir.symbols.SymbolTable`
         :param bool is_module_scope: whether or not the declarations are in \
                                      a module scoping unit. Default is False.
 
         :returns: the Fortran declarations for the table.
         :rtype: str
 
-        :raises VisitorError: if one of the symbols is a RoutineSymbol \
-            which does not have an ImportInterface or LocalInterface (and is \
-            not a Fortran intrinsic) as this is not supported by this backend.
+        :raises VisitorError: if one of the symbols is a RoutineSymbol which \
+            does not have an ImportInterface or ModuleDefaultInterface (and \
+            is not a Fortran intrinsic) as this is not supported by this \
```

Comment:
> if we see a Call to one then we could use that information

This is in fact what we do.

I forgot to mention UnresolvedInterface because I deal with them at the beginning (by letting them be undeclared if there are wildcard imports). But indeed we support RoutineSymbols with UnresolvedInterface, so I updated the docstring. Well spotted.